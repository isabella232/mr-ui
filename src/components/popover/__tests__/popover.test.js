import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './popover-test-cases';

describe('Popover', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.allProps.description, () => {
    beforeEach(() => {
      testCase = testCases.allProps;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
