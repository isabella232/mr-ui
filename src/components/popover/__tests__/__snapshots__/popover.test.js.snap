// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover all props renders 1`] = `
<Displaced
  alignment="top"
  allowPlacementAxisChange={false}
  containWithinViewport={false}
  getAnchorElement={[Function]}
  getContainingElement={[Function]}
  hasPointer={false}
  hideWhenAnchorIsOffscreen={true}
  mounted={true}
  onElement={[Function]}
  placement="left"
  pointerColor="red"
>
  <div
    aria-label="Title that is accessible"
    className="px12 py12"
    data-popover-id={2}
    data-test={true}
    key="body"
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="dialog"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
    tabIndex={-1}
  >
    <div
      className="w240 h240"
    >
      test popover
    </div>
  </div>
</Displaced>
`;

exports[`Popover basic renders 1`] = `
<Displaced
  alignment="top"
  allowPlacementAxisChange={true}
  containWithinViewport={true}
  getAnchorElement={[Function]}
  hasPointer={true}
  hideWhenAnchorIsOffscreen={false}
  mounted={true}
  onElement={[Function]}
  placement="right"
  pointerColor="#fff"
>
  <div
    className="round py6 px12 shadow-darken25"
    data-popover-id={1}
    key="body"
    onFocus={[Function]}
    onKeyDown={[Function]}
    role="dialog"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
    tabIndex={-1}
  >
    <div
      className="w240 h240"
    >
      test popover
    </div>
  </div>
</Displaced>
`;
